x-pack/test/api_integration/apis/ml/jobs/category_results.ts

```diff
@@ -0,0 +1,229 @@
+    ],
+    grok_pattern:
+      '.*?%{NUMBER:field}.+?failed.+?to.+?execute.+?bulk.+?item.+?index.+?index.+?testing-twitter-pycon-realtime.+?doc.+?%{NUMBER:field2}.+?source.+?n/a.+?actual.+?length.+?max.+?length.+?%{NUMBER:field3}.*',
+    num_matches: 1,
+    result_type: 'category_definition',
+    mlcategory: '3',
+  };
+
+  async function runTopCategoriesRequest(jobId: string, count = 5) {
+    const { body, status } = await supertest
+      .post(`/internal/ml/jobs/top_categories`)
+      .auth(USER.ML_POWERUSER, ml.securityCommon.getPasswordForUser(USER.ML_POWERUSER))
+      .set(getCommonRequestHeader('1'))
+      .send({ jobId, count });
+    ml.api.assertResponseStatusCode(200, status, body);
+
+    return body;
+  }
+
+  async function runGetCategoryRequest(jobId: string, categoryId: string) {
+    const { body, status } = await supertest
+      .get(`/internal/ml/anomaly_detectors/${jobId}/results/categories/${categoryId}`)
+      .auth(USER.ML_POWERUSER, ml.securityCommon.getPasswordForUser(USER.ML_POWERUSER))
+      .set(getCommonRequestHeader('1'));
+    ml.api.assertResponseStatusCode(200, status, body);
+
+    return body;
+  }
+
+  describe('Categorization job results', function () {
+    before(async () => {
+      await esArchiver.loadIfNeeded('x-pack/test/functional/es_archives/ml/categorization_small');
+      await ml.testResources.setKibanaTimeZoneToUTC();
+
+      await ml.api.createAnomalyDetectionJob(job);
+      await ml.api.createDatafeed(datafeed);
+      await ml.api.openAnomalyDetectionJob(job.job_id);
+      await ml.api.startDatafeed(catDatafeedId, { start: '0', end: String(Date.now()) });
+      await ml.api.waitForDatafeedState(catDatafeedId, DATAFEED_STATE.STOPPED);
+      await ml.api.waitForJobState(catJobId, JOB_STATE.CLOSED);
+    });
+
+    after(async () => {
+      await ml.api.cleanMlIndices();
+      await ml.testResources.cleanMLSavedObjects();
+    });
+
+    it('should have the correct top categories', async () => {
+      const result = await runTopCategoriesRequest(catJobId);
+      expect(result).to.eql(expectedTopCategories);
+    });
+
+    it('should get the correct category', async () => {
+      const result = await runGetCategoryRequest(catJobId, '3');
```

Comment:
Could we have a test to check the response when an invalid job ID or category ID are provided?